app/v3/controllers/RetrieveBalanceAndTransactionsController.scala

```diff
@@ -68,9 +71,11 @@ class RetrieveBalanceAndTransactionsController @Inject() (val authService: Enrol
         RequestHandler
           .withValidator(validator)
           .withService(service.retrieveBalanceAndTransactions)
+          .withModelHandling { response: RetrieveBalanceAndTransactionsResponse => documentDetailAdditionalField(response) }
```

Comment:
The controller has no real right to know intimate details about the model as it introduces an unnecessary coupling to something that is arguably a business logic concern (not for the controller i.e. HTTP/REST). 

We need to think how best to do this kind of thing going forward (calling from service layer seems a little more appropriate, and easier) but for now can we instead call a method on the RetrieveBalanceAndTransactionsResponse case class (something like `def adjustFields(featureSwitches: ...`). This hides some of the  details of the model from the controller (and allows polymorpic behaviour necessary for multi-schema).